packages/go_router/lib/go_router.dart

```diff
@@ -6,24 +6,284 @@
 /// deep linking, data-driven routes and more
 library go_router;
 
+import 'package:flutter/rendering.dart';
 import 'package:flutter/widgets.dart';
 
-import 'src/go_router.dart';
+import 'src/configuration.dart';
+import 'src/logging.dart';
+import 'src/matching.dart';
+import 'src/platform.dart';
+import 'src/routing.dart';
+import 'src/typedefs.dart';
 
-export 'src/custom_transition_page.dart';
-export 'src/go_route.dart';
-export 'src/go_router.dart';
-export 'src/go_router_refresh_stream.dart';
-export 'src/go_router_state.dart';
-export 'src/inherited_go_router.dart';
-export 'src/route_data.dart' show GoRouteData, TypedGoRoute;
-export 'src/typedefs.dart' show GoRouterPageBuilder, GoRouterRedirect;
-export 'src/url_path_strategy.dart';
+export 'src/configuration.dart' show GoRouterState, GoRoute;
+export 'src/misc/refresh_stream.dart';
+export 'src/pages/custom_transition_page.dart';
+export 'src/platform.dart' show UrlPathStrategy;
+export 'src/typed_routing.dart' show GoRouteData, TypedGoRoute;
+export 'src/typedefs.dart';
+
+/// The top-level go router class.
+///
+/// Create one of these to initialize your app's routing policy.
+// ignore: prefer_mixin
+class GoRouter extends ChangeNotifier with NavigatorObserver {
```

Comment:
What's the benefit of moving `GoRouter` to this file? Is it to cut down the number of files, or are there additional benefits?

I'm concerned that this forces things that depend on `GoRouter` to also be embedded in this file (like `InheritedGoRouter`), which may cause this file to bloat over time.